lib/Parser.js

```diff
@@ -1396,6 +1388,214 @@ class Parser extends Tapable {
+				this.prewalkFunctionDeclaration(node);
+				break;
+			case "IfStatement":
+				this.prewalkIfStatement(node);
+				break;
+			case "ImportDeclaration":
+				this.prewalkImportDeclaration(node);
+				break;
+			case "LabeledStatement":
+				this.prewalkLabeledStatement(node);
+				break;
+			case "SwitchStatement":
+				this.prewalkSwitchStatement(node);
+				break;
+			case "TryStatement":
+				this.prewalkTryStatement(node);
+				break;
+			case "VariableDeclaration":
+				this.prewalkVariableDeclaration(node);
+				break;
+			case "WhileStatement":
+				this.prewalkWhileStatement(node);
+				break;
+			case "WithStatement":
+				this.prewalkWithStatement(node);
+				break;
+		}
+	}
+
+	_walkNode(node) {
+		switch(node.type) {
+			case "ArrayExpression":
+				this.walkArrayExpression(node);
+				break;
+			case "ArrayPattern":
+				this.walkArrayPattern(node);
+				break;
+			case "ArrowFunctionExpression":
+				this.walkArrowFunctionExpression(node);
+				break;
+			case "AssignmentExpression":
+				this.walkAssignmentExpression(node);
+				break;
+			case "AssignmentPattern":
+				this.walkAssignmentPattern(node);
+				break;
+			case "AwaitExpression":
+				this.walkAwaitExpression(node);
+				break;
+			case "BinaryExpression":
+				this.walkBinaryExpression(node);
+				break;
+			case "BlockStatement":
+				this.walkBlockStatement(node);
+				break;
+			case "CallExpression":
+				this.walkCallExpression(node);
+				break;
+			case "CatchClause":
+				this.walkCatchClause(node);
+				break;
+			case "ClassDeclaration":
+				this.walkClassDeclaration(node);
+				break;
+			case "ClassExpression":
+				this.walkClassExpression(node);
+				break;
+			case "ConditionalExpression":
+				this.walkConditionalExpression(node);
+				break;
+			case "DoWhileStatement":
+				this.walkDoWhileStatement(node);
+				break;
+			case "ExportDefaultDeclaration":
+				this.walkExportDefaultDeclaration(node);
+				break;
+			case "ExportNamedDeclaration":
+				this.walkExportNamedDeclaration(node);
+				break;
+			case "ExpressionStatement":
+				this.walkExpressionStatement(node);
+				break;
+			case "ForInStatement":
+				this.walkForInStatement(node);
+				break;
+			case "ForOfStatement":
+				this.walkForOfStatement(node);
+				break;
+			case "ForStatement":
+				this.walkForStatement(node);
+				break;
+			case "FunctionDeclaration":
+				this.walkFunctionDeclaration(node);
+				break;
+			case "FunctionExpression":
+				this.walkFunctionExpression(node);
+				break;
+			case "Identifier":
+				this.walkIdentifier(node);
+				break;
+			case "IfStatement":
+				this.walkIfStatement(node);
+				break;
+			case "LabeledStatement":
+				this.walkLabeledStatement(node);
+				break;
+			case "LeftRightExpression":
+				this.walkLeftRightExpression(node);
+				break;
+			case "LogicalExpression":
+				this.walkLogicalExpression(node);
+				break;
+			case "MemberExpression":
+				this.walkMemberExpression(node);
+				break;
+			case "MethodDefinition":
```

Comment:
not covered by tests.

This is not entered via walkStatement, but via walkClass directly. Can be removed?

Reply:
Yup